### PR TITLE
docs(analytics): Desired analytics data

### PR DESCRIPTION
**Pull request by iliapolo**
This PR introduces a long lived analytics document that describes our analytics mechanism. 
Currently only includes which data we'd like to record, backed solely on client side capabilities of the app. 

The question still remains how to actually store the data, which will be addressed in a subsequent PR. 

Related to https://github.com/cdklabs/construct-hub-webapp/issues/8.

### CODE REVIEW

File: docs/analytics.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+Since we don't identify individual users, we approximate by looking at the total number of distinct visits to package pages, assuming
+that individual user traffic doesn't have a large variance.
+
+Note that in this case, we are not interested in users who stumble upon a package page during a search, but rather users who intentionally visit the package
+page to help them operate the construct library.
+
+**Required metrics:**
+
+- `visits.packages.<package>`: Number of distinct and direct visits to a `<package>` page.
+
```
[the remark this answers is not in the document]
Comment (MrArnoldPalmer) [reply docs/analytics.md:65]: I agree with this I think its not a good signal.

File: docs/analytics.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+**Why:** The overall experience with the hub should be positive.
+
+To answer this, we need to understand whether existing users are abandoning the hub, or keep using it.
+Normally this is done by making sure that the amount of daily active users doesn't decrease over time.
+Since we don't identify individual users, we approximate by looking at the total number of distinct visits to the hub, assuming
+that individual user traffic doesn't have a large variance.
+
+**Required metrics:**
+
+- `visits.hub`: Number of distinct visits to the hub. (main page + package pages)
```
Comment (MrArnoldPalmer) [inline docs/analytics.md:34]: I'd love a little summary section for `all required metrics` that shows them all in place just so we can easily look through them and compare.

**Comment by eladb** [comment]
Keep please

**Comment by iliapolo** [comment]
Keep

**Pull request closed without merging**